services/frontend-service/src/ui/components/CommitInfo/CommitInfo.test.tsx

```diff
@@ -209,6 +299,29 @@ test('CommitInfo component renders commit info when the response is valid', () =
         const actualCommitDescriptionTable = tables[0];
         const actualEventsTable = tables[1];
 
+        const targetElements = container.getElementsByClassName('history-button-container');
+        if (testCase.commitInfo.touchedApps.length < 2) {
+            if (testCase.commitInfo.previousCommitHash !== '' && testCase.commitInfo.nextCommitHash !== '') {
+                expect(targetElements.length).toEqual(2);
+                expect(targetElements[0]).toHaveTextContent('Previous Commit');
+                expect(targetElements[1]).toHaveTextContent('Next Commit');
+            } else {
+                if (testCase.commitInfo.previousCommitHash !== '' || testCase.commitInfo.nextCommitHash !== '') {
+                    const target = targetElements[0];
+                    if (testCase.commitInfo.previousCommitHash !== '') {
+                        expect(target).toHaveTextContent('Previous Commit');
+                    }
+                    if (testCase.commitInfo.nextCommitHash !== '') {
+                        expect(target).toHaveTextContent('Next Commit');
+                    }
+                } else {
+                    expect(targetElements.length).toEqual(0);
+                }
+            }
+        } else {
+            expect(targetElements.length).toEqual(0);
+        }
```

Comment:
The test should be more specific.
In general, we should avoid generic test functions like this `verifyTable`. Even though they have their place, they are not the ideal way to test most cases. Also `next/previous` are not part of the html table.

For this PR, I expect a specific test with it's own test array, and that test only looks at the `next` and `previous` button. There should also be very few `if` statements in tests.
(this is a bit hard to explain via text only)